### ISSUE TITLE
About Monkey patch ruby for net/http added verbosity in #13745

### ISSUE THREAD

**Issue by san983** [chef/chef#13936]
## Description

Since 18.3.0 merging and releasing https://github.com/chef/chef/pull/13745, these outputs started showing up as part of the monkey patch added in the given PR. Perhaps the `puts` should be moved to `Chef::Log.info` or `Chef::Log.debug`? 

## Chef Version
18.3.0

## Platform Version

https://github.com/chef/chef/pull/13745

## Replication Case

Just running chef-client in any affected env/platform

## Client Output

```
$ sudo chef-client --format min
Chef Infra Client, version 18.3.0
Patents: https://www.chef.io/patents
Infra Phase starting
opening connection to chef.example.com:443...
opened
starting SSL for chef.example.com:443...
SSL established, protocol: TLSv1.2, cipher: ECDHE-ECDSA-AES256-GCM-SHA384
opening connection to chef.example.com:443...
opened
starting SSL for chef.example.com:443...
SSL established, protocol: TLSv1.2, cipher: ECDHE-ECDSA-AES256-GCM-SHA384
opening connection to chef.example.com:443...
opened
...
```

## Stacktrace

N/A

**Comment by Tensibai** [comment]
I vote for `Chef::Log.debug` as in the net/http gem these are debug logs https://github.com/ruby/ruby/blob/v3_1_2/lib/net/http.rb#L1702

**Comment by 8la** [comment]
I vote for `Chef:Log.debug`, we will do a downgrade on our clients for a while, in order to make the debug process less annoying ; which  truly is, with this noise around the logs . 

Hope you can fix this soon.

Thanks!

**Comment by tpowell-progress** [comment]
This will catch the next Chef 18 promotion to stable. Still wrapping up Windows building issue but it will also be in any "current" channel release >= 18.3.0

**Comment by jkimalane** [comment]
Any news about removing those "puts" statements from http monkey patch?

**Comment by Stromweld** [comment]
This should be fixed in the next 18.4.x release that should be coming soon.

**Comment by jkimalane** [comment]
You should really focus more on QA as this issue is not the first one, which requires monkey patching a monkey patch

**Comment by tpowell-progress** [comment]
> You should really focus more on QA as this issue is not the first one, which requires monkey patching a monkey patch

Agreed, but that's an 🌲 problem in software engineering, compounded by the fact that there 20+ OS+architecture builds for each Chef version _and_ that, while extremely annoying, extraneous output to STDOUT is not in itself a functional breakage (and even in manual testing I can identify plenty of times where such obvious errors have slapped me in the face but I had tunnel vision on a different problem).

Perhaps we need a scaffolding to validate output... or maybe even fail automated tests if `STDOUT.puts` is called?  I think something like `expect(STDOUT).not_to receive(:puts)` could do that, but it would have to be only for a subset of test cases, since we have legitimate uses of `STDOUT.puts` in shell and some other areas.

Open to ideas from the community to detect similar classes of errors.

**Comment by Tensibai** [comment]
@jkimalane I find your comment a bit harsh in the form.

@tpowell-progress naive idea: have a dumb run with a simple file resource against zero server and expect its stdout to exactly match something curated by a human (excluding the first line with the client version which couldn't match obvisouly). This should catch any extraneous output but will likely need update when wording change somewhere in the codebase but should catch unwanted things. I guess this could be extended to solo without zero with and withou dry-run enabled.

For the record a workaround is that you can dump the fixed file in the linked PR in a "temp-fix" cookbook `libraries` directory to cut the extra output to only early calls (before the cookbook is loaded) for test purposes and have a recipe with a file resource updating the 18.4 client file in place while waiting for the next release.

**Comment by tpowell-progress** [comment]
@Tensibai I think breakage on wording changes is acceptable, as long as the test failure and description itself indicates that wording changes need to be checked as possible "false positives"

**Comment by gene1wood** [comment]
I can confirm this is fixed in 18.4.12 from #13970 
Can this issue be closed out now?

**Comment by rjhornsby** [comment]
So this was released. That is, the messages have been pushed to the logging handler at a `DEBUG` level. Forgive me, while this is better, only slightly better. A screen/page full of repetitive

```
[2024-06-14T17:32:08+00:00] DEBUG: opening connection to 169.254.169.254:80...
[2024-06-14T17:32:08+00:00] DEBUG: opened
```

provides no meaningful value and no useful diagnostic information. The same argument as previously applies - it's noise with no signal and interferes with reading relevant debug logs. I understand why someone might have needed to see this for their own troubleshooting, but it should have been removed before being committed (I've made this mistake myself), but ultimately not kept in the release.

Could we please remove these messages entirely?

![Screenshot 2024-06-14 at 12 39 45](https://github.com/chef/chef/assets/5077610/6699ad98-e885-42ec-b032-b0c927e06760)

**Comment by gene1wood** [comment]
@neha-p6 rjhornsby makes a good point about 

https://github.com/chef/chef/pull/13745/files#diff-eb7b1d76f85cee2774ee5d05119b1fdb53819293b3e03cc7e15b91eebb431c21R26

and

https://github.com/chef/chef/pull/13745/files#diff-eb7b1d76f85cee2774ee5d05119b1fdb53819293b3e03cc7e15b91eebb431c21R36

What are your thoughts on these debug lines?

**Comment by jaymzh** [comment]
Oops, this was closed because it had the waiting-on-release adn that release had happened, sorry.

**Comment by jaymzh** [comment]
There. Moved the one to trace and nuked the other. That work for ya?

**Comment by rjhornsby** [comment]
> There. Moved the one to trace and nuked the other. That work for ya?

IMHO I think that's probably good. I can see an argument for why someone might want the "connecting to" at the trace level. Thanks.

**Comment by gene1wood** [comment]
Oh whoops, I also just submitted #14959 to remove those lines. I'll close mine out in favor of #14958
